google-cloud-logging/src/test/java/com/google/cloud/logging/LoggingImplTest.java

```diff
@@ -1576,6 +1617,88 @@ public void testListResourceDescriptorAsyncWithOptions()
         Iterables.toArray(page.getValues(), MonitoredResourceDescriptor.class));
   }
 
+  @Test
+  public void testListLogs() {
+    EasyMock.replay(rpcFactoryMock);
+    logging = options.getService();
+    List<String> logNames = ImmutableList.of(LOG_NAME1, LOG_NAME2);
+    configureListLogsTests(logNames, LOG_NAMES_CURSOR);
+
+    Page<String> page = logging.listLogs();
+    assertEquals(LOG_NAMES_CURSOR, page.getNextPageToken());
+    assertArrayEquals(logNames.toArray(), Iterables.toArray(page.getValues(), String.class));
+  }
+
+  @Test
+  public void testListLogsEmpty() {
+    EasyMock.replay(rpcFactoryMock);
+    logging = options.getService();
+    List<String> emptyList = ImmutableList.of();
+    configureListLogsTests(emptyList, LOG_NAMES_CURSOR);
+
+    Page<String> page = logging.listLogs();
+    assertEquals(LOG_NAMES_CURSOR, page.getNextPageToken());
+    assertArrayEquals(emptyList.toArray(), Iterables.toArray(page.getValues(), String.class));
+  }
+
+  @Test
+  public void testListLogsNextPage() throws ExecutionException, InterruptedException {
+    EasyMock.replay(rpcFactoryMock);
+    logging = options.getService();
+    List<String> logNames1 = ImmutableList.of(LOG_NAME1, LOG_NAME2);
+    List<String> logNames2 = ImmutableList.of(LOG_NAME1);
+    String nextPageCursor = "nextCursor";
+    configureListLogsTests(logNames1, logNames2, LOG_NAMES_CURSOR, nextPageCursor);
```

Comment:
Readability-wise, it's hard to tell what these setup methods are doing (especially since it's an overloaded method).

Reply:
I will document the method and will try to improve the method name too :+1: